Sources/_CryptoExtras/Util/Error.swift

```diff
@@ -15,3 +15,11 @@
 public enum _CryptoRSAError: Error {
     case invalidPEMDocument
 }
+
+/// Errors that can be thrown when working with ``MLDSA``.
+public enum CryptoMLDSAError: Error {
```

Comment:
Let's not add a new error enum here, can we use our existing errors?

Reply:
Can we use `CryptoKitError.underlyingCoreCryptoError` when BoringSSL fails to generate a key or sign?

Reply:
Nevermind, I found `internalBoringSSLError()` 👍🏻